libkbfs/md_ops_test.go

```diff
@@ -62,85 +76,68 @@ func mdOpsShutdown(mockCtrl *gomock.Controller, config *ConfigMock) {
 }
 
-func addFakeRMDData(crypto cryptoPure, rmd *RootMetadata, h *TlfHandle) {
-	rmd.SetRevision(MetadataRevision(1))
-	rmd.SetLastModifyingWriter(h.FirstResolvedWriter())
-	rmd.SetLastModifyingUser(h.FirstResolvedWriter())
-	fakeVerifyingKey := MakeFakeVerifyingKeyOrBust("fake key")
-	rmd.SetWriterMetadataSigInfo(kbfscrypto.SignatureInfo{
-		Version:      kbfscrypto.SigED25519,
-		Signature:    []byte{41},
-		VerifyingKey: fakeVerifyingKey,
-	})
-
+func addFakeBRMDData(t *testing.T,
+	codec kbfscodec.Codec, crypto cryptoPure, brmd *BareRootMetadataV2,
+	h *TlfHandle) ExtraMetadata {
+	brmd.SetRevision(MetadataRevision(1))
+	pmd := PrivateMetadata{}
+	// TODO: Will have to change this for private folders if we
+	// un-mock out those tests.
+	buf, err := codec.Encode(pmd)
+	require.NoError(t, err)
+	brmd.SetSerializedPrivateMetadata(buf)
+	brmd.SetLastModifyingWriter(h.FirstResolvedWriter())
+	brmd.SetLastModifyingUser(h.FirstResolvedWriter())
+	var extra ExtraMetadata
 	if !h.IsPublic() {
-		rmd.FakeInitialRekey(crypto, h.ToBareHandleOrBust())
+		extra, err = brmd.FakeInitialRekey(crypto, h.ToBareHandleOrBust())
+		require.NoError(t, err)
 	}
+	return extra
 }
 
 func newRMD(t *testing.T, config Config, public bool) (
 	*RootMetadata, *TlfHandle) {
 	id := FakeTlfID(1, public)
 
 	h := parseTlfHandleOrBust(t, config, "alice,bob", public)
-	rmd := NewRootMetadata()
-	err := rmd.Update(id, h.ToBareHandleOrBust())
+	var brmd BareRootMetadataV2
+	err := brmd.Update(id, h.ToBareHandleOrBust())
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	addFakeRMDData(config.Crypto(), rmd, h)
+	extra := addFakeBRMDData(t, config.Codec(), config.Crypto(), &brmd, h)
 
-	return rmd, h
+	return &RootMetadata{
+		bareMd: &brmd,
+		extra:  extra,
+	}, h
 }
 
-func addFakeRMDSData(crypto cryptoPure, rmds *RootMetadataSigned, h *TlfHandle) ExtraMetadata {
-	rmds.MD.SetRevision(MetadataRevision(1))
-	rmds.MD.SetSerializedPrivateMetadata([]byte{1})
-	rmds.MD.SetLastModifyingWriter(h.FirstResolvedWriter())
-	rmds.MD.SetLastModifyingUser(h.FirstResolvedWriter())
-	fakeVerifyingKey := MakeFakeVerifyingKeyOrBust("fake key")
-	rmds.MD.SetWriterMetadataSigInfo(kbfscrypto.SignatureInfo{
-		Version:      kbfscrypto.SigED25519,
-		Signature:    []byte{41},
-		VerifyingKey: fakeVerifyingKey,
-	})
-	rmds.SigInfo = kbfscrypto.SignatureInfo{
-		Version:      kbfscrypto.SigED25519,
-		Signature:    []byte{42},
-		VerifyingKey: fakeVerifyingKey,
-	}
-	rmds.untrustedServerTimestamp = time.Now()
-
-	var extra ExtraMetadata
-	if !h.IsPublic() {
-		extra, _ = rmds.MD.FakeInitialRekey(crypto, h.ToBareHandleOrBust())
-	}
-	return extra
-}
+func newRMDS(t *testing.T, config Config, h *TlfHandle) (
+	*RootMetadataSigned, ExtraMetadata) {
+	id := FakeTlfID(1, h.IsPublic())
 
-func newRMDS(t *testing.T, config Config, public bool) (
-	*RootMetadataSigned, *TlfHandle, ExtraMetadata) {
-	id := FakeTlfID(1, public)
+	var brmd BareRootMetadataV2
+	// MDv3 TODO: uncomment the below when we're ready for MDv3
```

Comment:
What good would uncommenting it do?  It's not used below, is it?  Maybe the var name is supposed to be `brmd` and the comment should add that the line above should be deleted as well?